addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/CrossSiteScriptingScanRuleUnitTest.java

```diff
@@ -1668,6 +1672,309 @@ protected Response serve(IHTTPSession session) {
+                        }
+                        String response;
+                        if (name != null) {
+                            String script =
+                                    String.format(
+                                            "var f = { week: '1', bad: %s, phase: '2', };", name);
+                            response =
+                                    getHtml(
+                                            "InputInScript.html",
+                                            new String[][] {{"script", script}});
+                        } else {
+                            response = getHtml("NoInput.html");
+                        }
+                        return newFixedLengthResponse(response);
+                    }
+                });
+
+        HttpMessage msg = this.getHttpMessage(path + "?name=test");
+        this.rule.init(msg, this.parent);
+
+        // When
+        rule.scan();
+
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo("1,x:alert(1),y:1"));
+        assertThat(alertsRaised.get(0).getParam(), equalTo("name"));
+        assertThat(alertsRaised.get(0).getAttack(), equalTo("1,x:alert(1),y:1"));
+        assertThat(alertsRaised.get(0).getRisk(), equalTo(Alert.RISK_HIGH));
+        assertThat(alertsRaised.get(0).getConfidence(), equalTo(Alert.CONFIDENCE_MEDIUM));
+    }
+
+    @Test
+    void shouldNotAlertXssInJsObjectQuotedStringWithNoTags() throws HttpMalformedHeaderException {
+        // Given
+        String path = "/user/search";
+        nano.addHandler(
+                new NanoServerHandler(path) {
+
+                    @Override
+                    protected Response serve(IHTTPSession session) {
+                        String name = getFirstParamValue(session, "name");
+                        Source htmlSrc = new Source(name);
+                        // remove html tags from input
+                        List<Tag> tags = htmlSrc.getAllTags();
+                        for (Tag tag : tags) {
+                            name = name.replaceAll(tag.toString(), "");
+                        }
+                        name = name.replaceAll("\'", "");
+                        String response;
+                        if (name != null) {
+                            String script =
+                                    String.format(
+                                            "var f = { week: '1', bad: '%s', phase: '2', };", name);
+                            response =
+                                    getHtml(
+                                            "InputInScript.html",
+                                            new String[][] {{"script", script}});
+                        } else {
+                            response = getHtml("NoInput.html");
+                        }
+                        return newFixedLengthResponse(response);
+                    }
+                });
```

Comment:
Since this handler is used repeatedly you could extract a local private method that creates/returns it.

Reply:
The first 3 or 4 usages appear to be essentially the same with the exception of the value of "name" that is being used. If a method is extracted and the appropriate value for name is passed as a parameter to the method then things are simplified.

Reply:
I thought of implementing that but the method gets conditional statements to handle 
- single quotes around the "name"
```
 String.format( "var f = { week: '1', bad: '%s', phase: '2', };", name);
```
```
 String.format( "var f = { week: '1', bad: %s, phase: '2', };", name);
```

- removing HTML tags from input
```
// remove html tags from input
List<Tag> tags = htmlSrc.getAllTags();
for (Tag tag : tags) {
    name = name.replaceAll(tag.toString(), "");
}
```
I think the method gets complicated 🙂

Reply:
Okay, fair enough.